OpenEdXMobile/src/main/java/org/edx/mobile/module/storage/Storage.java

```diff
@@ -142,11 +144,35 @@ public int removeDownload(VideoModel model) {
     }
 
     public void removeAllDownloads() {
-        // Get all on going downloads
-        List<VideoModel> result = db.getListOfOngoingDownloads(null);
-        for (VideoModel model : result) {
-            removeDownload(model);
+        final String username = loginPrefs.getUsername();
+        final String sha1Username;
+        if (TextUtils.isEmpty(username)) {
+            return;
+        } else {
+            sha1Username = Sha1Util.SHA1(username);
         }
+
+        // Get all on going downloads
+        db.getListOfOngoingDownloads(new DataCallback<List<VideoModel>>(false) {
+            @Override
+            public void onResult(List<VideoModel> result) {
+                // Get all on going downloads and remove from db
+                long [] videoIds = new long[result.size()];
+                int i = 0;
+                for (VideoModel model : result) {
```

Comment:
Instead of a `for-each` loop, we can use the conventional `for` loop here for brevity.